Meadowlark-js/backends/meadowlark-postgresql-backend/test/integration/Read.test.ts

```diff
@@ -0,0 +1,113 @@
+  newDocumentInfo,
+  newSecurity,
+  documentIdForDocumentInfo,
+  GetRequest,
+  UpsertRequest,
+  NoResourceInfo,
+  ResourceInfo,
+  newResourceInfo,
+} from '@edfi/meadowlark-core';
+import { Client } from 'pg';
+import { resetSharedClient, getSharedClient } from '../../src/repository/Db';
+import { deleteAll } from '../../src/repository/Delete';
+import { getDocumentById } from '../../src/repository/Get';
+import { getRecordExistsSql } from '../../src/repository/QueryHelper';
+import { upsertDocument } from '../../src/repository/Upsert';
+
+jest.setTimeout(40000);
+
+const newGetRequest = (): GetRequest => ({
+  id: '',
+  resourceInfo: NoResourceInfo,
+  security: { ...newSecurity() },
+  traceId: 'traceId',
+});
+
+const newUpsertRequest = (): UpsertRequest => ({
+  id: '',
+  resourceInfo: NoResourceInfo,
+  documentInfo: NoDocumentInfo,
+  edfiDoc: {},
+  validate: false,
+  security: { ...newSecurity() },
+  traceId: 'traceId',
+});
+
+describe('given the get of a non-existent document', () => {
+  let client;
+  let getResult;
+
+  const resourceInfo: ResourceInfo = {
+    ...newResourceInfo(),
+    resourceName: 'School',
+  };
+  const documentInfo: DocumentInfo = {
+    ...newDocumentInfo(),
+    documentIdentity: [{ name: 'natural', value: 'get1' }],
+  };
+  const id = documentIdForDocumentInfo(resourceInfo, documentInfo);
+
+  beforeAll(async () => {
+    client = (await getSharedClient()) as Client;
+
+    getResult = await getDocumentById({ ...newGetRequest(), id, resourceInfo }, client);
+  });
+
+  afterAll(async () => {
+    await deleteAll(client);
+    await resetSharedClient();
+  });
+
+  it('should not exist in the db', async () => {
+    const result = await client.query(await getRecordExistsSql(id));
+
+    expect(result.rowCount).toBe(0);
+  });
+
+  it('should return get failure', async () => {
+    expect(getResult.response).toBe('GET_FAILURE_NOT_EXISTS');
+  });
+});
+
+describe('given the get of an existing document', () => {
+  let client;
+  let getResult;
+
+  const resourceInfo: ResourceInfo = {
+    ...newResourceInfo(),
+    resourceName: 'School',
+  };
+  const documentInfo: DocumentInfo = {
+    ...newDocumentInfo(),
+    documentIdentity: [{ name: 'natural', value: 'get2' }],
+  };
+  const id = documentIdForDocumentInfo(resourceInfo, documentInfo);
+
+  beforeAll(async () => {
+    client = (await getSharedClient()) as Client;
+    const upsertRequest: UpsertRequest = { ...newUpsertRequest(), id, documentInfo, edfiDoc: { inserted: 'yes' } };
+
+    // insert the initial version
+    await upsertDocument(upsertRequest, client);
+
+    getResult = await getDocumentById({ ...newGetRequest(), id, resourceInfo }, client);
+  });
+
+  afterAll(async () => {
+    await deleteAll(client);
+    await resetSharedClient();
+  });
+
+  it('should return the document', async () => {
+    expect(getResult.response).toBe('GET_SUCCESS');
+    expect(getResult.document.edfiDoc.inserted).toBe('yes');
```

Comment:
This isn't right. Your code is returning something like a `DocumentInfo` as `document` on the `GetResult`. However, the `document` _is_ the `edfiDoc` as defined on a `DocumentInfo`. The correct expectation should be `expect(getResult.document.inserted).toBe('yes');`

Reply:
Ah, got it. I added the values we stored in the database to the document, including the JSON edfi_doc but didn't realize that I needed to iterate over the JSON and add those as well.